runtime/compiler/env/VMJ9.h

```diff
@@ -777,27 +777,48 @@ class TR_J9VMBase : public TR_FrontEnd
     *    Return MemberName.vmtarget, a J9method pointer for method represented by `memberName`
     *    Caller must acquire VM access
     */
-   TR_OpaqueMethodBlock* targetMethodFromMemberName(uintptr_t memberName);
+   virtual TR_OpaqueMethodBlock* targetMethodFromMemberName(uintptr_t memberName);
    /*
     * \brief
     *    Return MemberName.vmtarget, a J9method pointer for method represented by `memberName`
     *    VM access is not required
     */
-   TR_OpaqueMethodBlock* targetMethodFromMemberName(TR::Compilation* comp, TR::KnownObjectTable::Index objIndex);
+   virtual TR_OpaqueMethodBlock* targetMethodFromMemberName(TR::Compilation* comp, TR::KnownObjectTable::Index objIndex);
    /*
     * \brief
     *    Return MethodHandle.form.vmentry.vmtarget, J9method for the underlying java method
     *    The J9Method is the target to be invoked intrinsically by MethodHandle.invokeBasic
     *    Caller must acquire VM access
     */
-   TR_OpaqueMethodBlock* targetMethodFromMethodHandle(uintptr_t methodHandle);
+   virtual TR_OpaqueMethodBlock* targetMethodFromMethodHandle(uintptr_t methodHandle);
    /*
     * \brief
     *    Return MethodHandle.form.vmentry.vmtarget, J9method for the underlying java method
     *    The J9Method is the target to be invoked intrinsically by MethodHandle.invokeBasic
     *    VM access is not required
     */
-   TR_OpaqueMethodBlock* targetMethodFromMethodHandle(TR::Compilation* comp, TR::KnownObjectTable::Index objIndex);
+   virtual TR_OpaqueMethodBlock* targetMethodFromMethodHandle(TR::Compilation* comp, TR::KnownObjectTable::Index objIndex);
+
+   /*
+    * \brief
+    *    Create and return a resolved method from member name index of an invoke cache array.
+    *    Encapsulates code that requires VM access to make JITServer support easier.
+    */
+   virtual TR_ResolvedMethod* targetResolvedMethodFromInvokeCacheArray(TR::Compilation *comp, TR_ResolvedMethod *owningMethod, uintptr_t *invokeCacheArray);
+
+   /*
+    * \brief
+    *    Return a Known Object Table index of an invoke cache array appendix element.
+    *    Encapsulates code that requires VM access to make JITServer support easier.
+    */
+   virtual TR::KnownObjectTable::Index getKnotIndexOfInvokeCacheArrayAppendixIndex(TR::Compilation *comp, uintptr_t *invokeCacheArray);
+
+   /*
+    * \brief
+    *    Wrapper for J9::SymbolReferenceTable::refineInvokeCacheElementSymRefWithKnownObjectIndex.
+    *    Encapsulates code that requires VM access to make JITServer support easier.
+    */
+   virtual TR::SymbolReference* refineInvokeCacheElementSymRefWithKnownObjectIndex(TR::Compilation *comp, TR::SymbolReference *originalSymRef, uintptr_t *invokeCacheArray, int32_t arrayIndex);
```

Comment:
Since we only ever refine appendix object symref when invokedynamic/invokehandle is resolved, we could change this to just handle refining the appendix object and rename the function accordingly. We can get rid of the `arrayIndex` parameter that remains the same in every call, and use the `JSR292_invokeCacheArrayAppendixIndex` macro directly in this function.

Reply:
Yeah, that was something I was wondering about. Are there no potential future uses, where we could pass a different index to this function? If not, then I agree with your suggestion

Reply:
No, we will not have to pass a different index in the future unless the side table design changes